tests/Hateoas/Tests/HateoasBuilderTest.php

```diff
@@ -118,14 +118,52 @@ public function testCyclicalReferences()
                 .'"name":"reference1",'
                 .'"_embedded":{'
                     .'"reference2":{'
-                        .'"name":"reference2",'
-                        .'"_embedded":{'
-                            .'"reference1":null'
-                        .'}'
+                        .'"name":"reference2",'.
+                        '"_embedded":{'.
+                            '"reference1":null'.
+                        '}'
                     .'}'
                 .'}'
             .'}',
-            $hateoas->serialize($reference1, 'json')
+            $hateoas->serialize($reference1, 'json', SerializationContext::create()->setSerializeNull(true))
+        );
+    }
+
+    public function testShouldNotSerialiseNull()
+    {
+        $hateoas = HateoasBuilder::create()->build();
+
+        $reference1 = new CircularReference1();
+        $reference2 = new CircularReference2();
+        $reference1->setReference2($reference2);
+        $reference2->setReference1($reference1);
+
+        $this->assertSame(
+            <<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<result>
+  <name><![CDATA[reference1]]></name>
+  <entry rel="reference2">
+    <name><![CDATA[reference2]]></name>
+    <entry rel="reference1"/>
```

Comment:
Why `reference1` is still mentioned here?

Reply:
ooops I need to change the XML too 👍

Reply:
Mmm actually it's not the same behaviour as embedded (JSON) serialisation.
Actually the entry is not null so I am not sure about how am I supposed to test it.
If you have any idea it would be much appreciate